File: backend/src/main/kotlin/application/services/AiModelService.kt
/*
 * ॐ नमः शिवाय
 * By: Sheersh Bhatnagar
 * Date: 24/04/26 17:10
*/

package dev.sheershbhatnagar.ai_assistant.application.services

import java.util.UUID

import dev.sheershbhatnagar.ai_assistant.domain.models.AiModel
import dev.sheershbhatnagar.ai_assistant.domain.repository.AiModelRepository

class AiModelService(private val aiModelRepository: AiModelRepository) {

    suspend fun addModelConfig(aiModel: AiModel): AiModel? {
        if (aiModel.apiKey.isBlank()) {
            throw IllegalArgumentException("API Key cannot be blank.")
        }

        return aiModelRepository.createModelConfig(aiModel)
    }

    suspend fun getUserModels(userId: UUID): List<AiModel> {
        return aiModelRepository.getModelsByUserId(userId)
    }

    suspend fun getModelDetails(modelId: UUID): AiModel? {
        return aiModelRepository.getModelById(modelId)
    }
}
    

File: backend/app/services/ai_model_service.py
from typing import List, Optional
from sqlalchemy.orm import Session
from app.models.ai_model import AiModel

class AiModelService:
    def add_model_config(self, db: Session, user_id: str, name: str, api_key: str) -> AiModel:
        if not api_key or api_key.strip() == "":
            raise ValueError("API Key cannot be blank.")

        new_model = AiModel(
            user_id=user_id,
            name=name,
            api_key=api_key
        )
        db.add(new_model)
        db.commit()
        db.refresh(new_model)
        return new_model

    def get_user_models(self, db: Session, user_id: str) -> List[AiModel]:
        return db.query(AiModel).filter(AiModel.user_id == user_id).all()

    def get_model_details(self, db: Session, model_id: str) -> Optional[AiModel]:
        return db.query(AiModel).filter(AiModel.id == model_id).first()
